packages/firestore/test/unit/local/query_engine.test.ts

```diff
@@ -508,7 +516,117 @@ function genericQueryEngineTest(
     );
     verifyResult(docs, [MATCHING_DOC_A]);
   });
-});
+
+  if (!durable) {
+    return;
+  }
+
+  if (!INDEXING_ENABLED) {
+    return;
+  }
+
+  it('combines indexed with non-indexed results', async () => {
```

Comment:
Should we port this to Android? I could not find it.

Reply:
I don't know why GitHub shows this as a new addition in the diff.

These tests exist in the tip-of-tree in the Web SDK. They also exist on Android. Sebastian ported these from Android to the Web via https://github.com/firebase/firebase-js-sdk/pull/6153